### DISCUSSION TITLE
Why does Quarto need to re-estimate and recreate the jupyter notebook every time a new format is rendered?

### DISCUSSION THREAD

**Discussion by friosavila** [quarto-dev/quarto-cli#4994] [Q&A] [answered]
Hi, I had a question regarding the use of Multiple Formats with Quarto and VSC.
Background: I'm working with my first document with Quarto and a Stata kernel. 

I was testing one of the new functions from Quarto. When creating multiple Formats, the HTML will also include links to all alternative formats. That is working quite well.

The question that I have, which may be more a suggestion for future updates, is why does Quarto need to re-estimate and recreate the jupyter notebook every time a new format is rendered. Since they are all based on the same ipynb, wouldn't it be more efficient to reuse the same one across all formats? 

Would it be the case that only happens with VSC? or because `Stata` (via nbstata) is not a formally supported language with Quarto?

Thank you
Fernando

**Comment by cscheid** [comment, marked as answer]
It's a (n unfortunately) general phenomenon. It happens because quarto can't know in general if your .ipynb file (or .qmd) has features that behave differently on different formats.

For example, some libraries generate completely different output depending on whether the output format is HTML or DocX or PDF.

If you want to avoid this, I recommend computing your results in a separate .ipynb and then using the `{{< embed >}}` shortcode with it explicitly.

**Comment by friosavila** [comment]
Thanks for the tip!
